### ISSUE TITLE
Rtabmap "Make" Error

### ISSUE THREAD

**Issue by Gabermonkey1** [introlab/rtabmap#777]
i've been trying to build Rtabmap on my raspberry pi 4, and it keeps giving me an error when i get to the make step:

pi@raspberrypi:~/rtabmap/build $ make
[ 80%] Built target rtabmap_utilite
[ 80%] Built target res_tool
[ 80%] Built target rtabmap_core
[ 80%] Built target rtabmap_gui
[ 80%] Linking CXX executable ../../../bin/rtabmap
/usr/bin/ld: /usr/lib/arm-linux-gnueabihf/libpcl_io.so: undefined reference to `libusb_set_option'
collect2: error: ld returned 1 exit status
make[2]: *** [app/src/CMakeFiles/rtabmap.dir/build.make:378: ../bin/rtabmap] Error 1
make[1]: *** [CMakeFiles/Makefile2:993: app/src/CMakeFiles/rtabmap.dir/all] Error 2
make: *** [Makefile:152: all] Error 2

I'm not any good with coding, or programming or anything of the sort, but i really want to start 3D scanning with Xbox Kinect 360 any help would be appreciated.

**Comment by matlabbe** [comment]
Hi, if it is only for testing simple kinect mapping, install the ros binaries instead. Here with [Ubuntu Mate 20.04](https://ubuntu-mate.org/download/arm64/):
```bash
# Setup ROS repo
sudo sh -c 'echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" > /etc/apt/sources.list.d/ros-latest.list'
sudo apt install curl # if you haven't already installed curl
curl -s https://raw.githubusercontent.com/ros/rosdistro/master/ros.asc | sudo apt-key add -
sudo apt update

# Install rtabmap
sudo apt install ros-noetic-rtabmap

# Launch:
rtabmap
```

Select Freenect driver. I tested on my RPI4, and it seems that freenect cannot always start the camera, rtabmap showing those errors:
```
[ERROR] (2021-07-21 16:57:53.905) CameraFreenect.cpp:156::getData() Not received any frames since 5 seconds, try to restart the camera again.
[ERROR] (2021-07-21 16:58:07.797) CameraFreenect.cpp:397::init() CameraFreenect: No devices connected!
[ERROR] (2021-07-21 16:58:20.153) CameraFreenect.cpp:156::getData() Not received any frames since 5 seconds, try to restart the camera again.
```
![Screenshot at 2021-07-21 17-01-12](https://user-images.githubusercontent.com/2319645/140192748-80761844-2b57-4894-8d7f-e8d0c1533aa2.jpg)

I had to deplug-replug the kinect in different USB of the RPI4 until it works:
![Screenshot at 2021-07-21 17-00-10](https://user-images.githubusercontent.com/2319645/140192703-3840eca5-e8bc-48e6-be24-46d5ab613634.jpg)

Note that performance is not super fast, so you have to move slowly.

**Comment by TheBricktop** [comment]
Will it work with the realsense d435?

**Comment by matlabbe** [comment]
Yes probably. If realsense-viewer works, rtabmap should work too.
